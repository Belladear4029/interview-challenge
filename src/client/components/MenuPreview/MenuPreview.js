// @flow
import React from "react";
import ItemCard from '../ItemCard/ItemCard'

type Props = {
  items: Array<{
    name: string,
    id: number,
    dietaries: Array<string>,
  }>,
  removeItem: Function
};

const MenuPreview = ({ items, removeItem }: Props): React$Element<React$FragmentType> => (
  <React.Fragment>
    <h2>Menu preview</h2>
    {items.length && (
      <ul className="menu-preview">
        {items.map((item) => (
          <li key={item.id} className="item">
            <ItemCard item={item} hasRemoveIcon={true} removeItem={removeItem} />
          </li>
        ))}
      </ul>
    )}
  </React.Fragment>
);

export default MenuPreview;
