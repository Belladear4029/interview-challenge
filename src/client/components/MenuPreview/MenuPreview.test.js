import * as React from "react";
import { shallow, mount } from "enzyme";
import MenuPreview from "./MenuPreview";
import { items } from "./mock.data";

const removeItem = () => {};

const shallowDefault = (props) =>
  shallow(<MenuPreview {...props} removeItem={(event) => removeItem(event)} />);
const mountDefault = (props) =>
  mount(<MenuPreview {...props} removeItem={(event) => removeItem(event)} />);

describe("<MenuPreview>", () => {
  it("renders correctly", () => {
    const wrapper = shallowDefault({ items });
    expect(wrapper).toMatchSnapshot();
  });
  it("renders correct props", () => {
    const wrapper = mountDefault({ items });
    expect(wrapper.props().items.length).toEqual(5);
  });
});
