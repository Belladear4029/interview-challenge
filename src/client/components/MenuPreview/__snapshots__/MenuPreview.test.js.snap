// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuPreview> renders correctly 1`] = `ShallowWrapper {}`;
